sdk/go/pulumi/resource.go

```diff
@@ -162,6 +204,12 @@ type Resource interface {
 
 	// addTransformation adds a single transformation to the resource.
 	addTransformation(t ResourceTransformation)
+
+	// markRemoteComponent marks this resource as a remote component resource.
+	markRemoteComponent()
+
+	// isRemoteComponent returns true if this is a local (i.e. in-process) component resource.
```

Comment:
This reads counter-intuitive to me. `isRemoteComponent=true` indicates this is a local component resource. Would it be more intuitive to have that isRemoteComponent=false`? Perhaps better naming alltogether?

Reply:
Ah, the comment is outdated. Sorry about the confusion here--the method name is accurate.